src/components/ButtonWithDropdownMenu.js

```diff
@@ -54,16 +54,26 @@ const ButtonWithDropdownMenu = (props) => {
     const [popoverAnchorPosition, setPopoverAnchorPosition] = useState(null);
     const {windowWidth, windowHeight} = useWindowDimensions();
     const caretButton = useRef(null);
-    useEffect(() => {
-        if (!caretButton.current) {
-            return;
-        }
+
+    const measurePopoverPosition = () => {
         caretButton.current.measureInWindow((x, y, w, h) => {
             setPopoverAnchorPosition({
                 horizontal: x + w,
                 vertical: y + h,
             });
+            setIsMenuVisible(true);
         });
+    };
+
+    useEffect(() => {
+        if (!caretButton.current) {
+            return;
+        }
+        if (popoverAnchorPosition !== null) {
+            measurePopoverPosition();
+        }
```

Comment:
Remove the null check. Also in the useEffect we should only measure the popover position, currently this function do both measure the position and show the menu.

Reply:
@s77rt  done

Reply:
actually I returned null check here because if we do not do it the popoverAnchorPosition is breaking again on the first render

Reply:
I'm not sure how would this break, you need to fix the other concern as well to get this working correctly

> Also in the useEffect we should only measure the popover position, currently this function do both measure the position and show the menu.

Reply:
@s77rt  yes, you are actually right, so I am adding an argument called showMenu to the measurePopoverPositionAndShowMenu function and set it to true when calling from onPress of caretButton so that it will call setIsMenuVisible(true) only when we click the dropdownButton while inside useEffect it will not call, please check it let me know if we can make anything else better here

Reply:
and also es-lint disabling rule is removed as well as we do not have popoverAnchorPosition inside useEffect anymore.